tests/s3/versioning/test_get_put_object_tagging.py

```diff
@@ -695,3 +695,236 @@ def test_get_put_obj_tags_susp_bkt_40432(self):
+                    self.object_name)
+        resp = s3_cmn_lib.get_object_tagging(s3_tag_test_obj=self.s3_tag_obj,
+                                             s3_ver_test_obj=self.s3_ver_obj,
+                                             bucket_name=self.bucket_name,
+                                             object_name=self.object_name)
+        assert_utils.assert_false(resp[0], resp)
+        assert_utils.assert_in("NoSuchKey", resp[1].message)
+
+        LOGGER.info("Step 13: Perform GET Object Tagging for %s with deletemarkerid=%s",
+                    self.object_name, dm_id)
+        resp = s3_cmn_lib.get_object_tagging(s3_tag_test_obj=self.s3_tag_obj,
+                                             s3_ver_test_obj=self.s3_ver_obj,
+                                             bucket_name=self.bucket_name,
+                                             object_name=self.object_name,
+                                             version_id=dm_id)
+        assert_utils.assert_true(resp[0], resp)
+        assert_utils.assert_in("NoSuchKey", resp[1].message)
+
+        LOGGER.info("Step 14: Perform PUT Object Tagging for %s with a tag key-value pair"
+                    " with deletemarkerid=%s", self.object_name, dm_id)
+        resp = s3_cmn_lib.put_object_tagging(s3_tag_test_obj=self.s3_tag_obj,
+                                             s3_ver_test_obj=self.s3_ver_obj,
+                                             bucket_name=self.bucket_name,
+                                             object_name=self.object_name,
+                                             version_tag=self.ver_tag, versions_dict=self.versions,
+                                             version_id=dm_id)
+        assert_utils.assert_false(resp[0], resp)
+        assert_utils.assert_in("NoSuchKey", resp[1].message)
+
+        LOGGER.info("ENDED: Test GET and PUT object tagging for deleted versioned object")
+
+    @pytest.mark.s3_ops
+    @pytest.mark.tags("TEST-40434")
+    def test_get_put_obj_tags_no_ver_40434(self):
+        """Test GET and PUT object tagging for non-existing version or object"""
+
+        LOGGER.info("STARTED: Test GET and PUT object tagging for non-existing version or object")
+
+        LOGGER.info("Step 1: Perform PUT Bucket versioning with status as Enabled on %s",
+                    self.bucket_name)
+        resp = self.s3_ver_obj.put_bucket_versioning(bucket_name=self.bucket_name)
+        assert_utils.assert_true(resp[0], resp)
+
+        LOGGER.info("Step 2: Upload Object %s with version enabled bucket %s",
+                    self.object_name, self.bucket_name)
+        s3_cmn_lib.upload_version(self.s3_test_obj, bucket_name=self.bucket_name,
+                                  file_path=self.file_path, object_name=self.object_name,
+                                  versions_dict=self.versions)
+        latest_ver_id1 = self.versions[self.object_name]["version_history"][-1]
+        latest_ver_id2 = "Vr1" * 9
```

Comment:
Let's rename this variable to something more descriptive eg. non_existing_version_id